apps/src/studio/studio.js

```diff
@@ -3327,18 +3337,31 @@ Studio.animateClouds = function() {
   }
 
   for (var i = 0; i < constants.NUM_CLOUDS; i++) {
+    Studio.cloudStep++;
+    var location = Studio.getCloudLocation(i);
     var cloud = document.getElementById('cloud' + i);
+    cloud.setAttribute('x', location.x);
+    cloud.setAttribute('y', location.y);
+  }
+};
+
 
-    var intervals = [ 50, 60 ];   // how many milliseconds to move a pixel
-    var distance = 700;           // how many pixels a cloud covers
+/** Gets the current location of a specified cloud.
+ * @param {number} cloudIndex
+ * @returns {Object} location
+ * @returns {number} location.x
+ * @returns {number} location.y
+ */
+Studio.getCloudLocation = function(cloudIndex) {
```

Comment:
maybe a few more comments here. distance is 700, but I thought width/height was 300 - not sure how they are different. Is one including the space between clouds?

The magic numbers of 100, 200, 400? Are those only meaningful because of the other values being 300 and 700?